server/src/main/java/org/apache/druid/segment/indexing/DataSchema.java

```diff
@@ -104,11 +106,13 @@ static void validateDatasourceName(String dataSource)
         !Strings.isNullOrEmpty(dataSource),
         "dataSource cannot be null or empty. Please provide a dataSource."
     );
-    Preconditions.checkArgument(!dataSource.contains("/"), "dataSource cannot contain the '/' character.");
+    Pattern p = Pattern.compile("(?s).*[^\\S ].*");
```

Comment:
May be worth using a private static named constant for this

Reply:
+1 for caching the compiled pattern. also why the pattern can't be simple `".*\s.*"` ?

Reply:
I've cached the compiled pattern, it can't be simple `".*\s.*"` because `\s` matches all whitespace including space, and we want to allow space.

Reply:
`[^\\S ]` matches all whitespace except space.

Reply:
Using the term "whitespace" in the code and PR title may be misleading as not all whitespace is forbidden

Reply:
thanks, for some reason I thought we did not allow space.